From 437008ef889e19d548a62003945d817ecb5d276a Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Wed, 24 Sep 2025 12:29:29 +0800
Subject: [PATCH 083/239] qml: silence clang-21 warnings

clang-21 emits -Wunnecessary-virtual-specifier when having virtual
functions in classes declared as `final`.

This patch removes these unnecessary `virtual` keywords and delcares the
formerly protected function private, as protected does not make sense
for final classes.

Change-Id: Ic6d91de5573f4b325e89bfbc2a9449791ccf94e2
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
---
 src/qml/qml/qqmlopenmetaobject_p.h | 5 ++---
 src/qmldom/qqmldomelements_p.h     | 3 ++-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/qml/qml/qqmlopenmetaobject_p.h b/src/qml/qml/qqmlopenmetaobject_p.h
index ce8d35ad3f..644a2d6155 100644
--- a/src/qml/qml/qqmlopenmetaobject_p.h
+++ b/src/qml/qml/qqmlopenmetaobject_p.h
@@ -47,10 +47,9 @@ public:
 
     QQmlPropertyCache::Ptr cache() const;
 
-protected:
-    virtual void propertyCreated(int, QMetaPropertyBuilder &);
-
 private:
+    void propertyCreated(int, QMetaPropertyBuilder &);
+
     QQmlOpenMetaObjectTypePrivate *d;
     friend class QQmlOpenMetaObject;
     friend class QQmlOpenMetaObjectPrivate;
diff --git a/src/qmldom/qqmldomelements_p.h b/src/qmldom/qqmldomelements_p.h
index 430cd34eb9..3a0471c6ff 100644
--- a/src/qmldom/qqmldomelements_p.h
+++ b/src/qmldom/qqmldomelements_p.h
@@ -879,7 +879,8 @@ public:
     void updatePathFromOwner(const Path &newPath) override;
     QString localDefaultPropertyName() const;
     QString defaultPropertyName(const DomItem &self) const;
-    virtual bool iterateSubOwners(const DomItem &self, function_ref<bool(const DomItem &owner)> visitor) const;
+    bool iterateSubOwners(const DomItem &self,
+                          function_ref<bool(const DomItem &owner)> visitor) const;
 
     QString idStr() const { return m_idStr; }
     QString name() const { return m_name; }
-- 
2.51.2

